lib/pages/chat/chat.dart

```diff
@@ -1172,6 +1171,20 @@ class ChatController extends State<Chat> with ImagePickerMixin, SendFilesMixin {
 
   @override
   Widget build(BuildContext context) => ChatView(this);
+
+  onUserScroll(UserScrollNotification notification) {
+    if (notification.direction == ScrollDirection.forward) {
+      if (!showTimestampBtn.value) {
+        setState(() => showTimestampBtn.value = true);
+      } else if (notification.direction == ScrollDirection.reverse) {
+        if (showTimestampBtn.value) {
+          setState(() => showTimestampBtn.value = false);
+        }
+      }
+    }
```

Comment:
I don't understand why you are checking `if (notification.direction == ScrollDirection.reverse)` inside `if (notification.direction == ScrollDirection.forward)`. Does it work as you expect?

Reply:
Nice catch, was just inattentive